Apps/WebClient/src/ClientApp/src/store/modules/user/actions.ts

```diff
@@ -241,4 +250,27 @@ export const actions: ActionTree<UserState, RootState> = {
                 });
         });
     },
+    getPatientData(context, params: { hdid: string }): Promise<void> {
```

Comment:
lets make an action that loads both userProfile and userPatientData. That one can be called everywhere else without having to coordinate what service it uses to load it.

Reply:
The better design would be checking if object is null in the getters.ts, then retrieve it, if not return it so that we don't have to worry about initializing a call just for getting data. Currently, we have to explicitly call the action at least once to get data.
This is how Singleton design pattern is recommended. But if we vote for the Singleton design pattern, we need to change our code for store.

Reply:
Added the check to load PatientData only once.

Reply:
Also the userProfile call may need to be called many time depending on the user registration status and user preference. Howerver, the userPatientData should be called only once. Getting them into a single action is not ideal.

Reply:
Vuex Getters are not meant to do logic to retrieve. If we want to retrieve on request we can use the Actions, That is part of the reason we check on the retrieve.

The reason for using getters vs actions is that Getters provide a reactive value. So the Vue class will automatically update if the getter value changes

Reply:
We can have a wrapper/manager for each Vuex (user, medication...) to ease the load, such as getting value from the Vuex Getters, if empty, do the Action to populate data so that we don't have to worry about which area in the application should retrieve data.

Reply:
I dont think that would be necessary since the action function will return a quick promise and we can only call the getter to retrieve the data.

Reply:
Btw, is OK if a single action fills both sets of data